### ISSUE TITLE
docs: Support for Other Language

### ISSUE THREAD

**Issue by CoolPlayLin** [faker-js/faker#2346]
### Clear and concise description of the problem

As a developer, I'd like docs would support for local language so that I could have better reading experience.

### Suggested solution

Create another repo to save translated docs. The translation work could leave it to contributors to do.

**Comment by xDivisionByZerox** [comment]
> [...] I could have better reading experience.

Thats interesting. As a german, I actually hate reading translated documentation. Often, most words remain in English due to the contextual relevance or if everything is translated it tends to sound unnatural and awkward. May I ask which translation languages you'd like to see in our documentation?

**Comment by CoolPlayLin** [comment]
@xDivisionByZerox 
Chinese. I think the transtlation work could hand over contributor who speak this as native language.If there is only English document, it may dissuade some users who are not familiar with English or use machine translation to get bad experiences

**Comment by ST-DDT** [comment]
We could potentially use a Crowdin or similar service to maintain the translations, that way we don't have to manage them in code.
As for putting the translated api docs in any kind of code bundle, I'm not sure whether that is possible at all.

@CoolPlayLin Are you willing to maintain the Chinese translations for at least a year, if we decide to add them?

(Currently, there doesn't seem to be that much of a need, if we look at the votes)

**Comment by CoolPlayLin** [comment]
@ST-DDT I'd like to if I have free time

**Comment by CoolPlayLin** [comment]
@ST-DDT I'm willing to do it if I have free time. I am going to maintain it almost every weekends.

Also, I think docs could add more version for different kinds of languages if there is someone willing to maintain it.

**Comment by rodalexanderson** [comment]
I also believe that open source projects will have more impact on community if they're translated to our own language. Here is my proposal for the changes in order to work this out. Definitely, we need someone to do the actual translation, I can do Spanish translation

So, the actual documentation pages are rendered by VitePress, and it uses de md files contained in the docs folder
The steps to follow are

1. We need to duplicate the actual folder inside docs

faker-js/
└── docs/
    ├── api/          <-- Original English content
    ├── guide/        <-- Original English content
    └── es/           <-- New folder for the Spanish translation
        ├── api/
        └── guide/
    └── zh/           <-- New folder for the Chinese translation
        ├── api/
        └── guide/

2. Someone would have to translate each document to their native language. These files don't change that much, please reference the following image

<img width="996" height="431" alt="Image" src="https://github.com/user-attachments/assets/f787831f-f87d-43ce-8efe-1cfb852498f6" />

3. Vite and vitepress have i18n integrated. The file docs/.vitepress/config.ts must be modified to add locales tp something like 

locales: {
  root: {
    lang: 'en-US',
    redirectFromRoot: true,
  },
  es: {
    label: 'Español',
    lang: 'es-ES',
    link: '/es/',
  },
},

A language switcher can be added in the faker-js/docs/.vitepress/config.ts

nav: [
      {
        text: 'Languages',
        items: [
          { text: 'English', link: '/' },
          { text: 'Español', link: '/es/' },
        ],

4. And for the maintenance, 
We could look at the differences in commits or create a small script to run git diff. Then we could be notified to translate the changes.

Like I said, I'm up for Spanish. Thanks!

**Comment by matthewmayer** [comment]
Would it make sense for each language to be in its own repo ?

Like github.com/@faker-js/faker-docs-es

Pros
- a script could be used to set up initial repo and sync any new docs changes
- these repos could have a lighter weight approval process which doesn't trigger full CI for every minor change. It would make it easier for someone to hack away on initial translation. 
- these repos could have different language-specific reviewers 

Cons
- We would likely need a more complex deployment script to generate the final multilingual docs site.

**Comment by Shinigami92** [comment]
I also thought about something similar, because when we e.g. do a major release, we (the maintainers) are writing an upgrade migration guide, and we wont like to get hold back and collect contributions for every language contribution to be in sync.

**Comment by ST-DDT** [comment]
Please keep in mind, that whatever usecase you have for translated docs also applies to the orginal ones (and vice versa).

- Want to update the translation? -> You might also want to fix a typo in the original docs.
- Is the translation/docs fix that important that it cannot wait a week for the next maintainance release? Haven't done any "content" changes to warrant a release => just instant push to the release branch as well (or use external sources)
- Both the original docs and the translated docs have to be versioned in parallel. next != v10 != v9
- If you change the original docs, you need a mechanism to invalidate the translations or notify relevant translators.
- Who and how are docs/translations reviewed? Who is (legally) responsible?

There exists tools for such things such as Crowdin. I recommend using such tools. Do NOT re-invent the wheel.

**Comment by matthewmayer** [comment]
I think some of the concerns could be addressed by having a banner at the top of translated docs explaining that these are community-maintained translated docs and may not be 100% up to date or accurate. "Don't let perfect be the enemy of good".

I think it would be good that if someone proposes setting up a new translation project that they find a team of at least 2-3 people who pledge to maintain the translations in the future.

**Comment by rodalexanderson** [comment]
I agree, @matthewmayer, they don't have to be perfect. Will look into gathering a team to continue support the project in Spanish
